go.mod

```diff
@@ -208,6 +209,8 @@ replace github.com/gogo/protobuf => github.com/gogo/protobuf v1.3.2
 // allows us to specify that as an option.
 replace google.golang.org/protobuf => github.com/lightninglabs/protobuf-go-hex-display v1.30.0-hex-display
 
+replace github.com/lightningnetwork/lnd/fn => ./fn
```

Comment:
Running `go mod tidy` doesn't generate a `go.sum`, this is because as is, `fn` has no deps outside the core standard library. Not sure we need to worry that much about CI here, since: as soon as this is merged, we'll tag, then land the PR that fixes PR by switching to the actual version. 

If we start to commit a `go.work` in the repo and use that in CI, then it would make things a bit easier, but unclear what the best practice of that actually is, as ref'd in #7958.